src/Cli/dotnet/commands/dotnet-tool/list/ToolListGlobalOrToolPathCommand.cs

```diff
@@ -54,6 +54,19 @@ public override int Execute()
                 toolPath = new DirectoryPath(toolPathOption);
             }
 
+            var packageEnumerable = GetPackages(toolPath, packageId);
+            PrintTable(packageEnumerable);
+
+            if (packageId.HasValue && !packageEnumerable.Any())
+            {
+                // return 1 if target package was not found
+                return 1;
+            }
+            return 0;
```

Comment:
It great to separate out the method. I think we could name the return value with a meaningful name.